core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/zookeeper.yml

```diff
@@ -0,0 +1,28 @@
+---
+- name: Include defaults from zookeeper role
+  include_vars:
+    file: roles/zookeeper/defaults/main.yml
+    name: zookeeper_defaults
+
+- name: Include pre-flight checks
+  include_tasks: zookeeper/preflight-check.yml
+
+- name: Get installed Zookeeper's version
+  shell: ls -la /opt/zookeeper | grep -oP '\d+\.\d+\.\d+'
```

Comment:
This is so ugly :disappointed:. The thing with `grep -o` is that there are different versions that behave differently on various distributions! I'd avoid using it. Please don't do `ls` in scripts, use `find` or even better here use the anisble `find:` module.

Reply:
Yes there is a difference between BSD grep and GNU grep in regards to the `-o` parameter, please avoid using it in production code! :scream:

Reply:
You are right, improvement done for that. "ls" and "grep" removed forever :)

Reply:
Fantastic improvement! :+1::heart_eyes: